[MERGE] dev-apa into master

AntonSkrub opened the pull request.

AntonSkrub committed:
[ADD] option to list branches and tags

AntonSkrub committed:
[UPD] added options to git operations

AntonSkrub committed:
[UPD] added cron job

- cron job for updating local copies based on a timed interval

AntonSkrub committed:
[UPD] moved logging info to separate file

- omitted logging commit history for logging latest commits on each branch

AntonSkrub committed:
[UPD] Improved some logs and variable names

AntonSkrub committed:
[FIX] removed redundant code

valentin-kaiser commented on pkg/goGit/goGit.go:
```diff
@@ -17,7 +14,7 @@ import (
 func UpdateLocalCopies(names []string, config *config.Config) {
 	for i := 0; i < len(names); i++ {
 		fmt.Printf("Value of i currently is: %v\n", i)
-		if i >= 3 {
+		if i >= 5 {
 			fmt.Println("we break here")
 			break
 		}
```
Remove the debug print. and why the if condition? If only for development purposes, I would remove these

AntonSkrub replied:
Resolved in [76b3090](https://github.com/AntonSkrub/GoGit-Integration/pull/6/commits/76b3090a4275bf0725703d0684da0ca154dd884f)

valentin-kaiser commented on pkg/config/config.go:
```diff
@@ -19,25 +19,26 @@ type Config struct {
 
-	EnableLog bool `yaml:"EnableLog"`
-	LogLevel  int  `yaml:"LogLevel"`
+	ListRefereces    bool `yaml:"ListRefereces"`
```
Misspelling: Refereces -> References

AntonSkrub replied:
Resolved in [76b3090](https://github.com/AntonSkrub/GoGit-Integration/pull/6/commits/76b3090a4275bf0725703d0684da0ca154dd884f)

valentin-kaiser commented on pkg/gitapi/gitapi.go:
```diff
@@ -41,16 +40,16 @@ func GetList(config *config.Config) []string {
 			logr.Errorf("[GitAPI] failed converting the repository name to string: %v\n", err)
 		} else {
 			repoNames = append(repoNames, name)
-			fmt.Printf("%v. %v%v\n", i, config.OrgaName, repo["name"])
```
The code is not relevant in any way. If you remove debug prints, please do it properly

AntonSkrub replied:
Resolved in [76b3090](https://github.com/AntonSkrub/GoGit-Integration/pull/6/commits/76b3090a4275bf0725703d0684da0ca154dd884f)

valentin-kaiser commented on pkg/goGit/log.go:
```diff
@@ -0,0 +1,94 @@
+	branchList := make([]string, 0)
+	tagList := make([]string, 0)
+
+	fmt.Println("------ remote branch references --------")
+	for _, ref := range refList {
+		refName := ref.Name().String()
+		if !strings.HasPrefix(refName, branchRefPrefix) {
+			if strings.HasPrefix(refName, tagRefPrefix) {
+				tagList = append(tagList, refName)
+			}
+			continue
+		}
+		branchList = append(branchList, refName)
+	}
+
+	for _, branch := range branchList {
+		branchName := branch[len(branchRefPrefix):]
+		fmt.Println(branchName)
+	}
+
+	fmt.Println("------ tag references --------")
+	if len(tagList) == 0 {
+		fmt.Println("--- No tags found ---")
+	} else {
+		for _, tag := range tagList {
+			tagName := tag[len(tagRefPrefix):]
+			fmt.Println(tagName)
+		}
+	}
```
Mixed logrus and fmt.Print usage. Unify to one of both

AntonSkrub replied:
Resolved in [76b3090](https://github.com/AntonSkrub/GoGit-Integration/pull/6/commits/76b3090a4275bf0725703d0684da0ca154dd884f)

AntonSkrub committed:
[FIX] Requested changes from PR #6

- fixed a typo
- renamed the module
- "renamed" pkg's
- edited logs
- removed unnecessarry prints

valentin-kaiser commented:
Will this be merged?

AntonSkrub committed:
[UPD] removed redundant code in log-funcs

fixed jscpd linter clone in log.go
